File: 240718/top k 숫자/top-k-elements.kt
import java.io.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val nums =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
    val set: List<Int> =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
            .toHashSet()
            .sorted()
    var count = 0
    for (i in set.size - 1 downTo 0) {
        if (count == nums[1])
            {
                break
            }
        print("${set.get(i)} ")
        count++
    }
}

File: 240718/top k 숫자/top-k-elements.js
const sortedSet = require('collections/sorted-set');
const fs = require('fs');

const input = fs.readFileSync(0).toString().trim().split("\n");
const [_, k] = input[0].trim().split(" ").map(Number);
const nums = input[1].trim().split(" ").map(Number);

const s = new sortedSet(nums);
let answer = [];

for (let i = 0; i < k; i++) {
    let g = s.findGreatest().value;
    answer.push(g);
    s.delete(g);
}

console.log(answer.join(" "));